### PR TITLE
Head off potential MDM bug

### PR DESCRIPTION
**Pull request by tadgh**
* Ensure any resource loaded as a golden resource is, in fact, golden. 
* Modify test for new error messages. 
* Add changelog

### CODE REVIEW

File: hapi-fhir-server-mdm/src/main/java/ca/uhn/fhir/mdm/provider/MdmControllerUtil.java [hunk 1]
```diff
@@ -57,7 +57,6 @@ public static IIdType extractSourceIdDtOrNull(String theName, String theSourceId
 
 	static IdDt getGoldenIdDtOrThrowException(String theParamName, String theId) {
 		IdDt goldenResourceId = new IdDt(theId);
-		//TODO GGG MDM: maybe add a gate here to only consider resources that can possibly be EMPI'ed?
 		if (goldenResourceId.getIdPart() == null) {
```
Comment (fil512) [inline hapi-fhir-server-mdm/src/main/java/ca/uhn/fhir/mdm/provider/MdmControllerUtil.java:60]: Adding comment in random location.

Please fix the debug log in `MdmResourceFilteringSvc#shouldBeProcessed`.  I noticed in a customer logfile it's printing our object references.  probably want the resource id here.

**Review by fil512 — APPROVED** [review]
Awesome

File: hapi-fhir-server-mdm/src/test/java/ca/uhn/fhir/mdm/provider/MdmControllerHelperTest.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class MdmControllerHelperTest {
+
+	@Test
+	void getLatestGoldenResourceFromIdOrThrowException() {
+
```
Comment (fil512) [inline hapi-fhir-server-mdm/src/test/java/ca/uhn/fhir/mdm/provider/MdmControllerHelperTest.java:11]: oops

File: hapi-fhir-server-mdm/src/main/java/ca/uhn/fhir/mdm/provider/MdmControllerUtil.java (hunk 1 above)
Comment (tadgh) [reply hapi-fhir-server-mdm/src/main/java/ca/uhn/fhir/mdm/provider/MdmControllerUtil.java:60]: Done in 6761267c729a78f23093a7c47461685541e9fcb6

File: hapi-fhir-server-mdm/src/test/java/ca/uhn/fhir/mdm/provider/MdmControllerHelperTest.java (hunk 2 above)
Comment (tadgh) [reply hapi-fhir-server-mdm/src/test/java/ca/uhn/fhir/mdm/provider/MdmControllerHelperTest.java]: fixed

**Pull request merged**